File: agent-plugin/plugin/src/main/java/org/matilda/template/TemplatePlugin.java
package org.matilda.template;

import org.matilda.commands.CommandRegistry;
import org.matilda.commands.PluginDependencies;
import org.matilda.commands.PluginDependenciesModule;

public class TemplatePlugin {
    public static CommandRegistry createCommandRegistry(PluginDependencies pluginDependencies) {
        return DaggerTemplatePluginComponent.builder()
                .pluginDependenciesModule(new PluginDependenciesModule(pluginDependencies))
                .build()
                .commandRegistry();
    }
}

File: agent-plugin/plugin/src/main/java/org/matilda/template/TemplatePlugin.kt
package org.matilda.template

import org.matilda.commands.CommandRegistry
import org.matilda.commands.PluginDependencies
import org.matilda.commands.PluginDependenciesComponent
import org.matilda.commands.create

object TemplatePlugin {
    @JvmStatic
    fun createCommandRegistry(pluginDependencies: PluginDependencies): CommandRegistry {
        return TemplatePluginComponent::class.create(PluginDependenciesComponent::class.create(pluginDependencies))
            .commandRegistry()
    }
}
